gensim/models/keyedvectors.py

```diff
@@ -311,11 +321,18 @@ def doesnt_match(self, words):
         """
         self.init_sims()
 
-        words = [word for word in words if word in self.vocab]  # filter out OOV words
-        logger.debug("using words %s" % words)
         if not words:
             raise ValueError("cannot select a word from an empty list")
-        vectors = vstack(self.syn0norm[self.vocab[word].index] for word in words).astype(REAL)
+        logger.debug("using words %s" % words)
+        vectors = []
+        for word in words:
+            try:
+                vectors.append(self.word_vec(word))
+            except KeyError:
+                logger.debug("vector for word %s not present, ignoring the word", word)
+        if not vectors:
+            raise ValueError("vector for all given words absent")
```

Comment:
Changed to a warning for now, that should be noticeable.
Not sure if an exception would be ideal, since it's a change in behaviour, and in case the intention is only to make things more transparent, a warning probably serves the purpose best.